babel/src/test/java/org/apache/calcite/test/BabelParserTest.java

```diff
@@ -974,6 +974,185 @@ private void checkParseInfixCast(String sqlType) {
+            + "return a + 1";
+    final String expected = "CREATE FUNCTION `ADD1` (`A` INTEGER) "
+        + "RETURNS INTEGER "
+        + "LANGUAGE SQL "
+        + "DETERMINISTIC "
+        + "COLLATION INVOKER INLINE TYPE 1 "
+        + "RETURN (`A` + 1)";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testCreateFunctionSqlFormCalledOnNUllInput() {
+    final String sql =
+        "create function add1(a Integer) "
+            + "returns Integer "
+            + "language sql "
+            + "called on null input "
+            + "collation invoker inline type 1 "
+            + "return a + 1";
+    final String expected = "CREATE FUNCTION `ADD1` (`A` INTEGER) "
+        + "RETURNS INTEGER "
+        + "LANGUAGE SQL "
+        + "CALLED ON NULL INPUT "
+        + "COLLATION INVOKER INLINE TYPE 1 "
+        + "RETURN (`A` + 1)";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testCreateFunctionSqlFormReturnsNullOnNUllInput() {
+    final String sql =
+        "create function add1(a Integer) "
+            + "returns Integer "
+            + "language sql "
+            + "returns null on null input "
+            + "collation invoker inline type 1 "
+            + "return a + 1";
+    final String expected = "CREATE FUNCTION `ADD1` (`A` INTEGER) "
+        + "RETURNS INTEGER "
+        + "LANGUAGE SQL "
+        + "RETURNS NULL ON NULL INPUT "
+        + "COLLATION INVOKER INLINE TYPE 1 "
+        + "RETURN (`A` + 1)";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testCreateFunctionSqlFormSpecificFunctionName() {
+    final String sql =
+        "create function add1(a Integer) "
+            + "returns Integer "
+            + "language sql "
+            + "specific someTable.plusOne "
+            + "collation invoker inline type 1 "
+            + "return a + 1";
+    final String expected = "CREATE FUNCTION `ADD1` (`A` INTEGER) "
+        + "RETURNS INTEGER "
+        + "LANGUAGE SQL "
+        + "SPECIFIC `SOMETABLE`.`PLUSONE` "
+        + "COLLATION INVOKER INLINE TYPE 1 "
+        + "RETURN (`A` + 1)";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testCreateFunctionSqlFormInvalidPositiveTypeIntFails() {
+    final String sql =
+        "create function foo() "
+            + "returns Integer "
+            + "language sql "
+            + "collation invoker inline type ^2^ "
+            + "return current_date + 1";
+    final String expected = "(?s).*Numeric literal.*out of range.*";
+    sql(sql).fails(expected);
+  }
+
+  @Test public void testCreateFunctionSqlFormInvalidNegativeTypeIntFails() {
+    final String sql =
+        "create function foo() "
+            + "returns Integer "
+            + "language sql "
+            + "collation invoker inline type ^-^2 "
+            + "return current_date + 1";
+    final String expected = "(?s).*Encountered \"-\" at .*";
+    sql(sql).fails(expected);
+  }
+
+  @Test public void testCreateFunctionSqlFormhasSecurityDefiner() {
```

Comment:
nit "Formhas" -> "FormHas"

Reply:
Fixes made in https://github.com/googleinterns/calcite/pull/79/commits/973cb3f514eabcc9dcb2e51bebf64c3d23d9e7c0 .